src/sidebar/components/annotation-user.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+
+const { isThirdPartyUser, username } = require('../util/account-id');
+const { withServices } = require('../util/service-context');
+
+/**
+ * Display information about an annotation's user. Link to the user's
+ * activity if it is a first-party user or `settings.usernameUrl` is present.
+ */
+function AnnotationUser({ annotation, features, serviceUrl, settings }) {
+  const user = annotation.user;
+  const isFirstPartyUser = !isThirdPartyUser(user, settings.authDomain);
+  const username_ = username(user);
+
+  // How should the user's name be displayed?
+  const displayName = (() => {
+    if (!isFirstPartyUser || !features.flagEnabled('client_display_names')) {
+      return username_;
```

Comment:
For third party users, the _display name_ should be rendered instead of the username. Consider the LMS context for example where the username is a randomly generated string.